### PR TITLE
feature: L2 notifier

### PR DESCRIPTION
**Pull request by dougmrqs**
This adds an example of implementation for contracts interested in receiving L2D, including the FIXED_RAND. Also adds some README instructions to give a high level view of how the contracts should behave.

Most of the code here is copied (but reviewed and adapted) from @avoloshko original work.

### CODE REVIEW

**Review by DZariusz — COMMENTED** [review]
I feel we need more comments to the code, so user understand more what is going on.

File: contracts/L2Notifier.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  constructor(address _registry) {
+    require(_registry != address(0x0), "_registry is empty");
+
+    registry = IRegistry(_registry);
+  }
+
+  /// @notice Register new delivery request, to future block
+  /// @param _key bytes32 encoded key
+  /// @param _minBlockId minimum Umbrella's sidechain block number to deliver data
+  function register(bytes32 _key, uint32 _minBlockId) external returns (bool success) {
```
Comment (dougmrqs) [inline contracts/L2Notifier.sol:28]: No need to return anything here, right? @avoloshko

File: contracts/L2Notifier.sol [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  /// @param _value value of the given key
+  /// @param _proof merkle proof of the layer 2 data
+  /// @dev This function already verifies Merkle Proof, so you don't need to verify it at receiver side
+  function notify(
+    address _subscriber,
+    uint32 _blockId,
+    bytes32 _key,
+    bytes32 _value,
+    bytes32[] memory _proof
+  ) external returns (bool success) {
```
Comment (dougmrqs) [inline contracts/L2Notifier.sol:58]: Neither here, right? @avoloshko

File: contracts/L2Notifier.sol (hunk 1 above)
Comment (avoloshko) [reply contracts/L2Notifier.sol:28]: Indeed, we should not return anything

File: contracts/L2Notifier.sol (hunk 2 above)
Comment (avoloshko) [reply contracts/L2Notifier.sol:58]: Indeed, we should not return anything

File: example.env [hunk 3]
```diff
@@ -1,8 +1,12 @@
-# RPC provider eg. infura
-BLOCKCHAIN_PROVIDER_URL=https://ropsten.infura.io/v3/***
+# RPC provider eg. infura or BSC Testnet
+#BLOCKCHAIN_PROVIDER_URL=https://ropsten.infura.io/v3/***
+BLOCKCHAIN_PROVIDER_URL=https://data-seed-prebsc-2-s3.binance.org:8545/
```
Comment (avoloshko) [inline example.env:3]: why would we comment out lines (#) in example.env?

**Comment by avoloshko** [comment]
@dougmrqs let's fix these minor comments and merge. Everything looks good

**Review by avoloshko — APPROVED** [review]

File: example.env (hunk 3 above)
Comment (dougmrqs) [reply example.env:3]: Removed.

**Pull request merged**